packages/beeq/src/components/progress/bq-progress.tsx

```diff
@@ -0,0 +1,157 @@
+
+  /** Progress bar thickness */
+  @Prop({ reflect: true }) thickness: TProgressThickness = 'medium';
+
+  /** Progress type */
+  @Prop({ reflect: true }) type: TProgressType = 'default';
+
+  /** It will set the border style of the progress bar */
+  @Prop({ reflect: true }) borderShape: TProgressBorderShape = 'rounded';
+
+  /** If `true`, a label text showing the value (in percentage) will be shown */
+  @Prop({ reflect: true }) label: boolean = false;
+
+  /** If `true`, a tooltip will be shown displaying the progress value */
+  @Prop({ reflect: true }) enableTooltip: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('borderShape')
+  @Watch('thickness')
+  @Watch('type')
+  handleTypePropChange() {
+    validatePropValue(PROGRESS_BORDER_SHAPE, 'rounded', this.el, 'borderShape');
+    validatePropValue(PROGRESS_THICKNESS, 'medium', this.el, 'thickness');
+    validatePropValue(PROGRESS_TYPE, 'default', this.el, 'type');
+  }
+
+  @Watch('value')
+  handleValuePropChange(newValue: number) {
+    this.validateValue(newValue);
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.handleTypePropChange();
+    this.handleValuePropChange(this.value);
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private validateValue(newValue: number) {
+    const clampedValue = Math.max(0, Math.min(100, newValue));
+    if (newValue !== clampedValue) {
+      this.value = clampedValue;
+    }
+    return this.value;
+  }
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    const style = {
+      ...(this.thickness === 'large' && { '--bq-progress-bar--height': '8px' }),
+      ...(this.type === 'error' && { '--bq-progress-bar--indicatorColor': 'var(--bq-ui--danger)' }),
+    };
+
+    return (
+      <Host style={style}>
+        <div class="flex items-center gap-xs">
+          <div class="relative flex w-full items-center">
+            <progress
+              class={{
+                [`progress-bar progress-bar__${this.type} ${this.thickness}`]: true,
+                'progress-bar__border-shape rounded-full': this.borderShape === 'rounded',
+              }}
+              value={this.indeterminate ? undefined : this.value}
+              max="100"
+            ></progress>
```

Comment:
You can use self close tag with JSX when there is no child

```suggestion
            <progress
              class={{
                [`progress-bar progress-bar__${this.type} ${this.thickness}`]: true,
                'progress-bar__border-shape rounded-full': this.borderShape === 'rounded',
              }}
              value={this.indeterminate ? undefined : this.value}
              max="100"
            />
```